ext/pcre/php_pcre.c

```diff
@@ -2664,6 +2664,33 @@ PHPAPI void php_pcre_split_impl(pcre_cache_entry *pce, zend_string *subject_str,
 }
 /* }}} */
 
+#define PREG_QUOTE_WORDLEN 32
+#define PREG_QUOTE_BITMASK (PREG_QUOTE_WORDLEN - 1)
+
+static const uint32_t preg_quote_bitmap[4] = { /* 128 bits ASCII map */
+	0x00000000, /* '\0' is handled differently */
+	0xf4006f1a, /* ! # $ ( ) * + - . : < = > ? */
+	0x78000000, /* [ \ ] ^ */
+	0x38000000  /* { | } */
+};
+
+static unsigned preg_quote_extra(char c)
+{/*{{{*/
+	unsigned u;
+	uint32_t w;
+
+	if (!isascii(c)) {
```

Comment:
You can accept unsigned char in this function and check `c >= 0x80` here (also obsoleting assert below). `is*` functions should be avoided due to potential locale dependence.

Reply:
I hesitated to do that and saw that the PHP code base makes use of assertions, so I figured `isascii()` would express the intent, and the assertion would map to the bitmap length.